### PR TITLE
Added Jokes, renamed Servlet to Funny Servlet, Gson to Json converter

### PR DESCRIPTION
**Pull request by Corramel**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/FunnyServlet.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+   */
+  private void addJokes() {
+      addJoke("Why did the chicken cross the road? To cross over.");
+      addJoke("Never play chess with a British person. Their queen never dies.");
+      addJoke("The Scottish military has eliminated the draft by issuing pants to all its officers.");
+      addJoke("The Scottish joke kilt me.");
+      addJoke("My pony can't neigh because he's a little hoarse.");
+      addJoke("If you don't know what an enigma is, and I don't know what an enigma is, then it's an absolute mystery.");
+      addJoke("Pressing F5 is so refreshing.");
+      addJoke("Oh. That moment when you realize who sang 'Take On Me'");
```
Comment (matty377) [inline portfolio/src/main/java/com/google/sps/servlets/FunnyServlet.java:44]: Ouch. This is a good one.

File: portfolio/src/main/java/com/google/sps/servlets/FunnyServlet.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+import com.google.gson.Gson;
+
+/** Handles request sent to /funny */
+@WebServlet("/funny")
+public class FunnyServlet extends HttpServlet {
+  private ArrayList<String> jokeArray = new ArrayList<String>();
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    if(jokeArray.isEmpty()) {
```
Comment (matty377) [inline portfolio/src/main/java/com/google/sps/servlets/FunnyServlet.java:19]: So, this is in memory on the servlet. It doesn't really persist between requests so you'll just be creating this array on every request (which is fine) and I don't think that's what you were trying to do.

You could just initialize a private static final List<String> and use that. I don't recall the syntax to do this off the top of my head but I'll find it for you.

It's all fine as-is, just wanted to clarify that state won't stick around for subsequent requests.

File: portfolio/src/main/webapp/servletScript.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+*/
+
+async function fetchString(){
+    const responseFromServer = await fetch('/funny');
+    const jsonFromServ = await responseFromServer.json();
+
+    console.log(jsonFromServ);
+
+    const linuxParagraph = document.getElementById("linux");
+    linuxParagraph.innerText = jsonFromServ[Math.floor(Math.random() * jsonFromServ.length)];
```
Comment (matty377) [inline portfolio/src/main/webapp/servletScript.js:12]: I don't recall the exact requirements of this week's task but you could potentially have a /funny/random servlet API that just picks a random joke and returns just a single string and does the randomization server-side.

No need to change anything, it's just useful to think about where you want logic to live in client-server applications. Neither option is more correct than the other in this case.

Side note: Why is the paragraphs ID "linux"? :)

**Review by matty377 — APPROVED** [review]
Nice job! I got a few good laughs from your jokes.

File: portfolio/src/main/webapp/servletScript.js (hunk 3 above)
Comment (Corramel) [reply portfolio/src/main/webapp/servletScript.js:12]: Yeah that's a good point, would it be faster that way? Though I think the task wanted us to do it this way. I forgot to change the IDs and I actually fixed it after I sent the commit

File: portfolio/src/main/java/com/google/sps/servlets/FunnyServlet.java (hunk 2 above)
Comment (Corramel) [reply portfolio/src/main/java/com/google/sps/servlets/FunnyServlet.java:19]: That's a good point, I think that's better actually.

**Pull request merged**